pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -254,6 +256,8 @@ async def run(
 
                 messages.append(model_response)
                 cost += request_cost
+                # TODO: is this the right location? Should we move this earlier in the logic?
+                execution_limit_settings.increment(request_cost)
```

Comment:
I personally would prefer if we added a `request_count` field to the `Cost` type, and then just did `execution_limit_settings.validate(cost)` here (rather than incrementing both cost and the limits).

I'd also prefer we rename `Cost` to `Usage` or similar, since that's really what it's representing now, and would make it feel less weird to add the `request_count` field. But even if we don't rename it like that, I think it's reasonable to add `request_count: int` (or `requests: int`) as a field on the type currently known as `Cost`